lectures/markov_chains.md

```diff
@@ -326,84 +312,105 @@ In these exercises, we'll take the state space to be $S = 0,\ldots, n-1$.
 (We start at $0$ because Python arrays are indexed from $0$.)
 
 
-### Rolling Our Own
+### Writing Our Own Simulation Code
+
+To simulate a Markov chain, we need 
 
-To simulate a Markov chain, we need its stochastic matrix $P$ and a probability mass function $\psi$ on $S$ from which to draw a realization of $X_0$.
+1. a stochastic matrix $P$ and 
+1. a probability mass function $\psi$ of length $n$ from which to draw a realization of $X_0$.
 
 The Markov chain is then constructed as follows:
 
-1. At time $t=0$, draw a realization of  $X_0$  from $\psi$.
+1. At time $t=0$, draw a realization of $X_0$ from the distribution $\psi$.
 1. At each subsequent time $t$, draw a realization of the new state $X_{t+1}$ from $P(X_t, \cdot)$.
 
 (That is, draw from row $X_t$ of $P$.)
 
-To implement this simulation procedure, we need a method for generating draws from a discrete distribution.
+To implement this simulation procedure, we need a method for generating draws
+from a discrete distribution.
 
 For this task, we'll use `random.draw` from [QuantEcon](http://quantecon.org/quantecon-py).
 
 To use `random.draw`, we first need to convert the probability mass function
 to a cumulative distribution
 
-TODO -- link to distributions lecture
-
 ```{code-cell} ipython3
 ψ = (0.3, 0.7)           # probabilities over {0, 1}
 cdf = np.cumsum(ψ)       # convert into cumulative distribution
 qe.random.draw(cdf, 5)   # generate 5 independent draws from ψ
 ```
 
++++ {"user_expressions": []}
+
 We'll write our code as a function that accepts the following three arguments
 
 * A stochastic matrix `P`
-* An initial state `init`
+* An initial distribution `ψ`
 * A positive integer `sample_size` representing the length of the time series the function should return
 
 ```{code-cell} ipython3
-def mc_sample_path(P, ψ_0=None, sample_size=1_000):
+def mc_sample_path(P, ψ=None, sample_size=1_000):
 
     # set up
     P = np.asarray(P)
     X = np.empty(sample_size, dtype=int)
 
     # Convert each row of P into a cdf
     n = len(P)
-    P_dist = [np.cumsum(P[i, :]) for i in range(n)]
+    P_dist = np.cumsum(P, axis=1)  # Convert rows into cdfs
 
     # draw initial state, defaulting to 0
-    if ψ_0 is not None:
-        X_0 = qe.random.draw(np.cumsum(ψ_0))
+    if ψ is not None:
+        X_0 = qe.random.draw(np.cumsum(ψ))
```

Comment:
Hi @jstac,

Thanks for reminding me about the consistency in notation. It helps a lot to keep the code clear.

I just noticed that you were avoiding the use of ψ_0 in this function for initial distribution. I can see why this is preferred given no other $\psi$s are used, so _0 is redundant, but I also think it would be good if the notation for initial distributions is treated the same throughout this lecture. Please kindly let me know what you think about changing all initial distributions to ψ_0.

Many thanks in advance.

Reply:
Hi @HumphreyYang , I changed that one to be consistent with the mathematics above.

But you are right --- the code is now inconsistent with code that follows.  Shall we change it back and then change the maths above it to use $\psi_0$ instead of $\psi$?

Reply:
BTW, I appreciate you raising this.  Getting all the small things right is essential to creating a great lecture.

Reply:
> Hi @HumphreyYang , I changed that one to be consistent with the mathematics above.
> 
> But you are right --- the code is now inconsistent with code that follows. Shall we change it back and then change the maths above it to use ψ0 instead of ψ?

Many thanks @jstac; I think changing to $ψ_0$ would be great. I will create a PR to check and change the math notation.